common/src/main/java/whocraft/tardis_refined/client/screen/waypoints/WaypointListScreen.java

```diff
@@ -55,44 +64,79 @@ protected void init() {
         super.init();
 
         setEvents(() -> {
-            if(tardisNavLocation != TardisNavLocation.ORIGIN) {
-                new TravelToWaypointMessage(tardisNavLocation.getName()).send();
+            if(waypoint != null) {
+                new TravelToWaypointMessage(waypoint.getId()).send();
                 Minecraft.getInstance().setScreen(null);
             }
             Minecraft.getInstance().setScreen(null);
         }, new SelectionScreenRun() {
             @Override
             public void onPress() {
-                if (tardisNavLocation != TardisNavLocation.ORIGIN) {
-                    new RemoveWaypointEntryMessage(tardisNavLocation.getName()).send();
+                if (waypoint != null) {
+                    new RemoveWaypointEntryMessage(waypoint.getId()).send();
                 }
             }
         });
 
 
-        addSubmitButton(width / 2 + 85, (height) / 2 - 35);
-        addTrashIcon(width / 2 + 85, (height) / 2 - 55);
+        SpriteIconButton spriteButtonNew = this.addRenderableWidget(CommonTRWidgets.imageButton(20, Component.translatable("Submit"), (arg) -> {
```

Comment:
Use a more meaningful variable name. Declare as a field if this is being reused like the load, edit button etc,

Reply:
Addressed